dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/utils/ProcessUtils.java

```diff
@@ -115,13 +115,23 @@ public static boolean kill(@NonNull TaskExecutionContext request) {
             // Get all child processes
             String pids = getPidsStr(processId);
             String[] pidArray = PID_PATTERN.split(pids);
-            if (pidArray.length == 0) {
+            if (StringUtils.isBlank(pids) || pidArray.length == 0) {
                 log.warn("No valid PIDs found for process: {}", processId);
                 return true;
             }
 
             // Convert PID string to list of integers
-            List<Integer> pidList = Arrays.stream(pidArray).map(Integer::parseInt).collect(Collectors.toList());
+            List<Integer> pidList = Arrays.stream(pidArray).filter(StringUtils::isNotBlank)
+                    .map(s -> {
+                        try {
+                            return Integer.parseInt(s.trim());
+                        } catch (NumberFormatException e) {
+                            log.warn("Invalid PID string ignored: {}", s);
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
```

Comment:
It's better to change `getPidsStr` return a validate pid list, rather than parse the pid in up layer.